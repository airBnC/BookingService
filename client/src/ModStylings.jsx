import styled from 'styled-components';
import logo from './images/arrowLogo.png';
import arrow from './images/arrow.png';

export const abnbLightGrey = '#F0F0F0';
export const abnbGrey = 'rgb(172, 172, 172)';
export const abnbRed = '#F16664';
export const abnbLightBlue = '#79CCCD';
export const abnbBlue = '#33DACD';
export const abnbCalendarBlue = '#00a699';
export const abnbDarkBlue = '#007D8C';

export const HolderMod1 = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center; 
  align-items: center;
  width: inherit;
  border-bottom: 1px solid ${abnbGrey};
  font-weight: bold;
`;

export const Holder = styled.div`
  display: flex;
  flex-direction: ${props => (props.mod1 ? 'row' : 'column')};
  width: inherit;
  padding-top: ${props => (props.mod1 ? '' : '10px')};
  justify-content: ${props => (props.mod1 ? 'flex-start' : '')};
`;

export const HolderMod4 = styled.div`
  display: flex;
  width: inherit;
  font-size: 13px;
  font-weight: ${props => (props.total ? 'bold' : '')};
  border-bottom: ${props => ((!props.main && !props.total) ? '1px solid rgb(172,172,172)' : '')};
  padding: 3px;
  padding-top: ${props => (props.main ? '17px' : '5px')};
  flex-direction: ${props => (props.main ? 'column' : 'row')};
  justify-content: ${props => (props.main ? '' : 'space-between')};
`;

export const InputHolder = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  border: 1px solid ${abnbGrey};
  padding: 5px;
  padding-left: 15px;
  padding-right: 15px;
  &:hover{
    cursor: pointer;
  }
`;

export const Title = styled.div`
  font-size: 12px;
  font-weight: bold;
  margin-bottom: 3px;
`;

export const Details = styled.div`
  font-size: 12px;
  margin-top: 12px;
`;

export const Logo = styled.div`
  content: url(${logo});
  height: 17px;
  width: 25px;
  transform: rotate(${props => (props.clicked ? '0turn' : '0.5turn')});
`;

export const Arrow = styled.div`
  content: url(${arrow});
  height: 17px;
`;

export const DropdownHolder = styled.div`
  position: absolute;
  z-index: 2;
  display: flex;
  margin-top: ${props => (props.guest ? '50px' : '')};
  flex-direction: column;
  width: 286px;
  border: 1px solid ${abnbGrey};
  border-radius: 1%;
  padding: 20px;
  background-color: white;
  user-select: none;
`;

export const Close = styled.button`
  margin-top: 8px;
  border: none;
  align-self: flex-end;
  font-size: 14px;
  color: ${abnbDarkBlue};
  outline: none;
  &:hover{
    text-decoration: underline;
  }
  &:active{
    color: ${abnbRed};
  }
`;
