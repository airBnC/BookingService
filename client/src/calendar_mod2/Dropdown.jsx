import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import CheckInCalendar from './CheckInCalendar';
import CheckOutCalendar from './CheckOutCalendar';
import arrow from '../images/arrow.png';

export const Holder = styled.div`
  position: absolute;
  z-index: 2;
  display: flex;
  flex-direction: column;
  width: 286px;
  border: 1px solid rgb(172, 172, 172);
  border-radius: 1%;
  padding: 20px;
  background-color: white;
`;

export const Arrow = styled.div`
  border-radius: 10%;
  content: url(${arrow});
  height: 20px;
  border: 1px solid rgb(172, 172, 172);
  border-radius: 10%;
  padding: 3px;
  transform: ${props => (props.left ? 'rotate(0.5turn)' : '')};
    &:active{
      background-color: #79CCCD;
    }
    &:hover {
      cursor: pointer;
    }
`;

export const SectionOne = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 10px;
  font-size: 18px;
  font-weight: bold;
`;

export const SectionTwo = styled.div`
  display: flex;
  align-items: center;
`;

export const SectionThree = styled.div`
  font-size: 12px;
  margin-top: 12px;
  display: flex;
  flex-direction: column;
`;

export default function Dropdown(props) {
  let calendar;
  let holderArrow;
  if (props.clicked === 1) {
    calendar = (<CheckInCalendar {...props} />);
    holderArrow = (
      <svg height="10" width="250" fill="rgb(172, 172, 172)" stroke="rgb(172, 172, 172)" position="absolute">
        <path d="M 15,25 35, 25 25, 0 Z" height="0px" position="absolute" />
      </svg>);
  } else if (props.clicked === -1) {
    calendar = (<CheckOutCalendar {...props} />);
    holderArrow = (
      <svg height="10" width="250" fill="rgb(172, 172, 172)" stroke="rgb(172, 172, 172)">
        <path d="M 0,25 35, 25 25, 15 Z" height="0px" />
      </svg>);
  }
  if (props.clicked === 1 || props.clicked === -1) {
    return (
      <div>
        {holderArrow}
        <Holder>
          <SectionOne>
            <Arrow left />
            <div> {props.month} {props.year} </div>
            <Arrow />
          </SectionOne>
          <SectionTwo>
            {calendar}
          </SectionTwo>
          <SectionThree>
            <div> {props.minStay} day minimum. </div>
            <div> Updated 2 days ago </div>
          </SectionThree>
        </Holder>
      </div>
    );
  }
  return null;
}

Dropdown.propTypes = {
  clicked: PropTypes.number,
  month: PropTypes.string,
  year: PropTypes.number,
  dateClick: PropTypes.func,
  // dates: PropTypes.arrayOf( PropTypes.shape( {
  //   date: PropTypes.date,
  // } ) ),

};
Dropdown.defaultProps = {
  clicked: 0,
  month: 'June',
  year: 2018,
  dateClick: () => null,
  // dates: [],
};
