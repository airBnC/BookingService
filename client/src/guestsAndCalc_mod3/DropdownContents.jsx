import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const GuestHolder = styled.div`
  display: flex;
  flex-direction: row;
`;

const GuestDetails = styled.div`
`;

const DownButton = styled.button`
`;

const UpButton = styled.button`
`;

export default function DropdownContents( props ) {
  DropdownContents.propTypes = {
    adults: PropTypes.number,
    childrens: PropTypes.number,
    infants: PropTypes.number,
    btnClick: PropTypes.func,
  };
  DropdownContents.defaultProps = {
    adults: 0,
    childrens: 0,
    infants: 0,
    btnClick: () => null,
  };

  const thingsToRender = [ 'adults', 'children', 'infants' ];
  const toRender = thingsToRender.map( thing =>
    ( <GuestHolder>
      <div> {thing} </div>
      <DownButton />
      <div> {props.adults} </div>
      <UpButton />
      </GuestHolder> ) );

  return (
    <div>
      {toRender}
    </div>

  );
}
