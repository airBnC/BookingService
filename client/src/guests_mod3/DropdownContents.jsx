import React from 'react';
import PropTypes from 'prop-types';
import { Holder, FeatureHolder, Guest, GuestDetails, ButtonDown, ButtonUp, Number } from './DropdownContentsStyling';

const list = ['adults', 'children', 'infants'];
const desc = ['', 'Ages 2-12', 'Under 2'];

export default function DropdownContents(props) {
  const toRender = list.map((propName, index) => {
    const guest = propName.replace(/s|ren/g, '');
    const guestNum = props[guest];
    return (
      <Holder key={propName}>
        <FeatureHolder>
          <Guest guest={propName}> {propName} </Guest>
          <GuestDetails> {desc[index]} </GuestDetails>
        </FeatureHolder>
        <FeatureHolder button guest={propName}>
          <ButtonDown
            disabled={(guest === 'adult' && guestNum === 1) || guestNum === 0}
            onClick={() => props.btnClick(propName, -1)}
            number={guestNum}
            guest={propName}
          >-
          </ButtonDown>
          <Number> {guestNum} </Number>
          <ButtonUp
            disabled={(guest === 'infant' && guestNum === 5) || (props.totalGuests === props.maxGuests && (guest === 'adult' || guest === 'child'))}
            onClick={() => props.btnClick(propName, 1)}
            number={guestNum}
            total={props.totalGuests}
            max={props.maxGuests}
            guest={propName}
          >+
          </ButtonUp>
        </FeatureHolder>
      </Holder>
    );
  });

  return (
    <div>
      {toRender}
    </div>

  );
}

DropdownContents.propTypes = {
  // btnClick: PropTypes.func,
  // adult: PropTypes.number,
  // child: PropTypes.number,
  // infant: PropTypes.number,
  totalGuests: PropTypes.number,
  maxGuests: PropTypes.number,

};
DropdownContents.defaultProps = {
  // btnClick: () => null,
  // adult: 0,
  // child: 0,
  // infant: 0,
  totalGuests: 0,
  maxGuests: 0,
};
