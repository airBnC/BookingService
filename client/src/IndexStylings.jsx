import styled, { injectGlobal } from 'styled-components';
import { abnbGrey, abnbRed } from './ModStylings';

injectGlobal([`
  html, body{
    height: 100%;
    width: 100%
  }
  body {
    display: flex;
    justify-content: center;
    align-items: center;
  }
`]);

export const Holder = styled.div`
  user-select: none;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 328px;  
  height: ${props => ((props.startDate && props.endDate) ? '400px' : '300px')};
  border: 1px solid ${abnbGrey};
  font-family: Roboto, sans-serif;
  padding: 24px;
`;

export const Button = styled.button`
  width: inherit;
  margin-top: 20px;
  height: 40px;
  border-radius: 7px;
  outline: none;
  background-color: ${abnbRed};
  font-weight: bold;
  color: white;
`;

export const Details = styled.div`
  font-size: 12px;
  font-weight: bold;
  align-self: center;
  margin-top: 7px;
`;
