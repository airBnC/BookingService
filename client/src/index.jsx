import React, { Component } from 'react';
import { render } from 'react-dom';
import styled, { injectGlobal } from 'styled-components';
import axios from 'axios';
import ModOne from './pricing_mod1/ModOne';
import ModTwo from './calendar_mod2/ModTwo';
import ModThree from './guests_mod3/ModThree';
import { setStartOrEndDate, clearDates, calendarChange } from './calendar_mod2/CalendarLogic';

injectGlobal([`
  html, body{
    height: 100%;
    width: 100%
  }
  body {
    display: flex;
    justify-content: center;
    align-items: center;
  }
`]);

export const Holder = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 328px;  
  height: 300px;
  border: 1px solid rgb(172, 172, 172);
  font-family: Quicksand;
  padding: 24px;
`;

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      adults: 1,
      children: 0,
      infants: 0,
      totalGuests: 1,
      listingData: null,
      startDate: null,
      endDate: null,
      mod2Clicked: 'closed',
      mod3Clicked: false,
    };
    this.setStartOrEndDate = setStartOrEndDate.bind(this);
    this.clearDates = clearDates.bind(this);
    this.calendarChange = calendarChange.bind(this);
  }

  componentDidMount() {
    this.getListingData();
  }

  onGuestButtonClick = (guest, increment) => {
    const val = this.state[guest] + increment;
    const { totalGuests } = this.state;
    const total = totalGuests + increment;
    const { maxGuests } = this.state.listingData;
    // infants don't count towards the number of guests, but have a cap of 5
    if (guest === 'infants' && (val > 5 || val < 0)) {
      return;
    // there is a minimum of 1 adult guest
    } else if (guest === 'adults' && (val > maxGuests || val < 1 || total > maxGuests)) {
      return;
    } else if (guest === 'children' && (val > maxGuests || val < 0 || total > maxGuests)) {
      return;
    }
    this.setState({
      [guest]: val,
      totalGuests: guest === 'infants' ? totalGuests : total,
    });
  }

  onMod3InputHolderClick = () => {
    this.setState({
      mod3Clicked: !this.state.mod3Clicked,
    });
  }

  getListingData = () => {
    let id = window.location.pathname;
    if (id === '/') {
      id = 1;
    } else {
      id = id.replace(/\//g, '');
    }
    axios.get(`http://127.0.0.1:3002/rooms/${id}/bookingInfo/`)
      .then((response) => {
        this.setState({
          listingData: response.data,
        });
      })
      .catch(() => {
        console.log('there was an error!');
      });
  }

  render() {
    if (this.state.listingData) {
      return (
        <Holder>
          <ModOne
            price={this.state.listingData.pricePerNight}
            rating={this.state.listingData.starRating}
            numReviews={this.state.listingData.custRevNum}
          />
          <ModTwo
            dates={this.state.listingData.datesTaken}
            minStay={this.state.listingData.minStay}
            startDate={this.state.startDate}
            endDate={this.state.endDate}
            setDate={this.setStartOrEndDate}
            calendarChange={this.calendarChange}
            clicked={this.state.mod2Clicked}
            clearDates={this.clearDates}
          />
          <ModThree
            adult={this.state.adults}
            child={this.state.children}
            infant={this.state.infants}
            totalGuests={this.state.totalGuests}
            btnClick={this.onGuestButtonClick}
            maxGuests={this.state.listingData.maxGuests}
            clicked={this.state.mod3Clicked}
            close={this.onMod3InputHolderClick}
          />
          {/* <ModFour /> */}
        </Holder>
      );
    }
    return null;
  }
}

render(<App />, document.getElementById('root'));

// cleaningFee={this.state.listingData.cleaningFee}
// maxGuests={this.state.listingData.maxGuests}
// serviceFee={this.state.listingData.serviceFee}
