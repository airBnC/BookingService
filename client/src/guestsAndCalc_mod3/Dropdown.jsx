import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import DropdownContents from './DropdownContents';

const Holder = styled.div`
  display: flex;
  flex-direction: column;
  width: 230px;
  border: 1px solid rgb(172, 172, 172);
  padding: 10px;
  padding-top: 0px;
`;

const Details = styled.div`
  font-size: 12px;
  margin-top: 10px;
`;

export default function Dropdown( props ) {
  Dropdown.propTypes = {
    clicked: PropTypes.bool,
    maxGuests: PropTypes.number,
    adult: PropTypes.number,
    child: PropTypes.number,
    infant: PropTypes.number,
    btnClick: PropTypes.func,
  };
  Dropdown.defaultProps = {
    clicked: false,
    maxGuests: 0,
    adult: 0,
    child: 0,
    infant: 0,
    btnClick: () => null,
  };
  if ( props.clicked ) {
    console.log( 'I was clicked!' );
    return (
      <Holder>
        <DropdownContents
          adult={props.adult}
          child={props.child}
          infant={props.infant}
          btnClick={props.btnClick}
        />
        <Details> {props.maxGuests } guests maximum. Infants don't count toward the number of guests. </Details>
      </Holder>
    );
  }
  return null;
}
