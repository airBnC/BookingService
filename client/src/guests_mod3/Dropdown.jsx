import React from 'react';
import PropTypes from 'prop-types';
import DropdownContents from './DropdownContents';
import { DropdownHolder, Close, Details } from '../ModStylings';

export default function Dropdown(props) {
  if (props.clicked) {
    return (
      <DropdownHolder guest>
        <DropdownContents
          {...props}
        />
        <Details>
          {props.maxGuests } guests maximum.
          Infants donâ€™t count toward the number of guests.
        </Details>
        <Close onClick={props.close}> Close </Close>
      </DropdownHolder>
    );
  }
  return null;
}

Dropdown.propTypes = {
  clicked: PropTypes.bool,
  btnClick: PropTypes.func,
  close: PropTypes.func,
  adult: PropTypes.number,
  child: PropTypes.number,
  infant: PropTypes.number,
  totalGuests: PropTypes.number,
  maxGuests: PropTypes.number,

};
Dropdown.defaultProps = {
  clicked: false,
  btnClick: () => null,
  close: () => null,
  adult: 0,
  child: 0,
  infant: 0,
  totalGuests: 0,
  maxGuests: 0,
};
