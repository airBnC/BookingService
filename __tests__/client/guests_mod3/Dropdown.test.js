import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import 'jest-styled-components';
import { configure, shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { DropdownHolder } from '../../../client/src/ModStylings';
import DropdownContents from '../../../client/src/guests_mod3/DropdownContents';
import Dropdown from '../../../client/src/guests_mod3/Dropdown';

configure({ adapter: new Adapter() });

const props = {
  clicked: true,
  btnClick: () => null,
  close: () => null,
  adult: 0,
  child: 0,
  infant: 0,
  totalGuests: 0,
  maxGuests: 0,
};

describe('Holder Component', () => {
  it('is the main flex component', () => {
    const tree = renderer
      .create(<DropdownHolder> things inside </DropdownHolder>)
      .toJSON();
    expect(tree).toHaveStyleRule('display', 'flex');
    expect(tree).toMatchSnapshot();
  });
});
describe('Dropdown Component', () => {
  it('renders the right amount of children', () => {
    const dropdown = mount(<Dropdown {...props} />);
    expect(dropdown.find(DropdownContents)).toHaveLength(1);
    expect(dropdown).toMatchSnapshot();
  });
  it('does not render if clicked is false', () => {
    const dropdown = mount(<Dropdown clicked={false} />);
    expect(dropdown).toMatchSnapshot();
  });
});
