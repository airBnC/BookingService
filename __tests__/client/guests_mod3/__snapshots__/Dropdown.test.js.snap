// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component does not render if clicked is false 1`] = `
<Dropdown
  adult={0}
  btnClick={[Function]}
  child={0}
  clicked={false}
  close={[Function]}
  infant={0}
  maxGuests={0}
  totalGuests={0}
/>
`;

exports[`Dropdown Component renders the right amount of children 1`] = `
.c11 {
  font-size: 12px;
  margin-top: 12px;
}

.c0 {
  position: absolute;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 50px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 286px;
  border: 1px solid rgb(172,172,172);
  border-radius: 1%;
  padding: 20px;
  background-color: white;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c12 {
  margin-top: 8px;
  border: none;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  font-size: 14px;
  color: #007D8C;
  outline: none;
}

.c12:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c12:active {
  color: #F16664;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: 10px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-transform: capitalize;
  margin-top: 10px;
  font-weight: bold;
}

.c8 {
  text-transform: capitalize;
  margin-top: 0px;
  font-weight: bold;
}

.c4 {
  font-size: 12px;
  min-height: 12px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 0px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: 4px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: 0px;
}

.c7 {
  min-width: 9px;
  font-size: 17px;
  margin-top: 2px;
}

.c6 {
  border-radius: 50%;
  height: 25px;
  width: 25px;
  margin-left: 7px;
  margin-right: 7px;
  margin-top: 1px;
  outline: none;
  border: 1px solid rgb(172,172,172);
}

.c6:active {
  background-color: #007D8C;
}

.c6:hover:enabled {
  cursor: pointer;
}

.c10 {
  border-radius: 50%;
  height: 25px;
  width: 25px;
  margin-left: 7px;
  margin-right: 7px;
  margin-top: 1px;
  outline: none;
  border: 1px solid #007D8C;
}

.c10:active {
  background-color: #007D8C;
}

.c10:hover:enabled {
  cursor: pointer;
}

<Dropdown
  adult={0}
  btnClick={[Function]}
  child={0}
  clicked={true}
  close={[Function]}
  infant={0}
  maxGuests={0}
  totalGuests={0}
>
  <ModStylings__DropdownHolder
    guest={true}
  >
    <div
      className="c0"
    >
      <DropdownContents
        adult={0}
        btnClick={[Function]}
        child={0}
        clicked={true}
        close={[Function]}
        infant={0}
        maxGuests={0}
        totalGuests={0}
      >
        <DropdownContentsStyling__Holder
          key="adults"
        >
          <div
            className="c1"
          >
            <DropdownContentsStyling__FeatureHolder>
              <div
                className="c2"
              >
                <DropdownContentsStyling__Guest
                  guest="adults"
                >
                  <div
                    className="c3"
                  >
                     
                    adults
                     
                  </div>
                </DropdownContentsStyling__Guest>
                <DropdownContentsStyling__GuestDetails>
                  <div
                    className="c4"
                  >
                     
                     
                  </div>
                </DropdownContentsStyling__GuestDetails>
              </div>
            </DropdownContentsStyling__FeatureHolder>
            <DropdownContentsStyling__FeatureHolder
              button={true}
              guest="adults"
            >
              <div
                className="c5"
              >
                <DropdownContentsStyling__Button
                  disabled={true}
                  guest="adults"
                  number={0}
                  onClick={[Function]}
                >
                  <button
                    className="c6"
                    disabled={true}
                    onClick={[Function]}
                  >
                    -
                  </button>
                </DropdownContentsStyling__Button>
                <DropdownContentsStyling__Number>
                  <div
                    className="c7"
                  >
                     
                    0
                     
                  </div>
                </DropdownContentsStyling__Number>
                <DropdownContentsStyling__Button
                  disabled={true}
                  guest="adults"
                  max={0}
                  number={0}
                  onClick={[Function]}
                  total={0}
                >
                  <button
                    className="c6"
                    disabled={true}
                    max={0}
                    onClick={[Function]}
                  >
                    +
                  </button>
                </DropdownContentsStyling__Button>
              </div>
            </DropdownContentsStyling__FeatureHolder>
          </div>
        </DropdownContentsStyling__Holder>
        <DropdownContentsStyling__Holder
          key="children"
        >
          <div
            className="c1"
          >
            <DropdownContentsStyling__FeatureHolder>
              <div
                className="c2"
              >
                <DropdownContentsStyling__Guest
                  guest="children"
                >
                  <div
                    className="c8"
                  >
                     
                    children
                     
                  </div>
                </DropdownContentsStyling__Guest>
                <DropdownContentsStyling__GuestDetails>
                  <div
                    className="c4"
                  >
                     
                    Ages 2-12
                     
                  </div>
                </DropdownContentsStyling__GuestDetails>
              </div>
            </DropdownContentsStyling__FeatureHolder>
            <DropdownContentsStyling__FeatureHolder
              button={true}
              guest="children"
            >
              <div
                className="c9"
              >
                <DropdownContentsStyling__Button
                  disabled={true}
                  guest="children"
                  number={0}
                  onClick={[Function]}
                >
                  <button
                    className="c6"
                    disabled={true}
                    onClick={[Function]}
                  >
                    -
                  </button>
                </DropdownContentsStyling__Button>
                <DropdownContentsStyling__Number>
                  <div
                    className="c7"
                  >
                     
                    0
                     
                  </div>
                </DropdownContentsStyling__Number>
                <DropdownContentsStyling__Button
                  disabled={true}
                  guest="children"
                  max={0}
                  number={0}
                  onClick={[Function]}
                  total={0}
                >
                  <button
                    className="c6"
                    disabled={true}
                    max={0}
                    onClick={[Function]}
                  >
                    +
                  </button>
                </DropdownContentsStyling__Button>
              </div>
            </DropdownContentsStyling__FeatureHolder>
          </div>
        </DropdownContentsStyling__Holder>
        <DropdownContentsStyling__Holder
          key="infants"
        >
          <div
            className="c1"
          >
            <DropdownContentsStyling__FeatureHolder>
              <div
                className="c2"
              >
                <DropdownContentsStyling__Guest
                  guest="infants"
                >
                  <div
                    className="c8"
                  >
                     
                    infants
                     
                  </div>
                </DropdownContentsStyling__Guest>
                <DropdownContentsStyling__GuestDetails>
                  <div
                    className="c4"
                  >
                     
                    Under 2
                     
                  </div>
                </DropdownContentsStyling__GuestDetails>
              </div>
            </DropdownContentsStyling__FeatureHolder>
            <DropdownContentsStyling__FeatureHolder
              button={true}
              guest="infants"
            >
              <div
                className="c9"
              >
                <DropdownContentsStyling__Button
                  disabled={true}
                  guest="infants"
                  number={0}
                  onClick={[Function]}
                >
                  <button
                    className="c6"
                    disabled={true}
                    onClick={[Function]}
                  >
                    -
                  </button>
                </DropdownContentsStyling__Button>
                <DropdownContentsStyling__Number>
                  <div
                    className="c7"
                  >
                     
                    0
                     
                  </div>
                </DropdownContentsStyling__Number>
                <DropdownContentsStyling__Button
                  disabled={false}
                  guest="infants"
                  max={0}
                  number={0}
                  onClick={[Function]}
                  total={0}
                >
                  <button
                    className="c10"
                    disabled={false}
                    max={0}
                    onClick={[Function]}
                  >
                    +
                  </button>
                </DropdownContentsStyling__Button>
              </div>
            </DropdownContentsStyling__FeatureHolder>
          </div>
        </DropdownContentsStyling__Holder>
      </DropdownContents>
      <ModStylings__Details>
        <div
          className="c11"
        >
          0
           guests maximum. Infants don’t count toward the number of guests.
        </div>
      </ModStylings__Details>
      <ModStylings__Close
        onClick={[Function]}
      >
        <button
          className="c12"
          onClick={[Function]}
        >
           Close 
        </button>
      </ModStylings__Close>
    </div>
  </ModStylings__DropdownHolder>
</Dropdown>
`;

exports[`Holder Component is the main flex component 1`] = `
.c0 {
  position: absolute;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 286px;
  border: 1px solid rgb(172,172,172);
  border-radius: 1%;
  padding: 20px;
  background-color: white;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  className="c0"
>
   things inside 
</div>
`;
